Assets/Scripts/Scriptables/Systems/DamageResistanceSystem.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Celeritas.Game;
+using Celeritas.Game.Entities;
+using Celeritas.Scriptables.Interfaces;
+using Sirenix.OdinInspector;
+using UnityEngine;
+
+namespace Celeritas.Scriptables.Systems
+{
+
+	/// <summary>
+	/// Reduces incoming damage by set percentage from all sources.
+	/// </summary>
+
+	[CreateAssetMenu(fileName = "New Ship Damage Resistance Modifier", menuName = "Celeritas/Modifiers/Ship Damage Resistance")]
+
+	public class DamageResistanceSystem : ModifierSystem, IEntityEffectAdded, IEntityEffectRemoved
+	{
+
+		[SerializeField, PropertyRange(-100, 100), Title("Damage Taken Modifier Percentage")]
+		private int amount;
+
+		[SerializeField, PropertyRange(-100, 100), Title("Damage Taken Modifier Percentage added per level")]
+		private int amountExtraPerLevel;
+
+		/// <summary>
+		/// The amount this system modifies damage taken
+		/// </summary>
+		public int Amount { get => amount; }
+
+		/// <summary>
+		/// How much the damage taken modifier increases per level
+		/// </summary>
+		public int AmountExtraPerLevel { get => amountExtraPerLevel; }
+
+		public override bool Stacks => true;
+
+		public override SystemTargets Targets => SystemTargets.Ship;
+
+		public override string GetTooltip(ushort level) => $"<color=green>▲</color> Reduces amount of damage taken by <color=green>{(Amount + (AmountExtraPerLevel * level)):0}%</color>.";
+
+		public void OnEntityEffectAdded(Entity entity, ushort level)
+		{
+			var ship = entity as ShipEntity;
+			ship.damageModifierPercentage = (Amount + (AmountExtraPerLevel * level));
+		}
+
+		public void OnEntityEffectRemoved(Entity entity, ushort level)
+		{
+			var ship = entity as ShipEntity;
+			ship.damageModifierPercentage = 0;
```

Comment:
Other systems might be changing this value in the future (or this system might be if it is stacked, or included in a different module.) If this does happen, removing 1 copy of the system will revert all other systems, even if they haven't been removed. Probably better to not set it to 0; perhaps subtract the amount you originally added?

Reply:
Yep, I see where this could go wrong.
I've applied the change to subtract the added amount.